### PR TITLE
Simplify the code for filtering find refs results and testing them.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

File: src/EditorFeatures/Test2/FindReferences/FindReferencesTests.DelegateInvokeMethodSymbols.vb [hunk 1]
```diff
@@ -17,9 +17,9 @@ Imports System.Linq
  
 Module Program
     Function Main(args As String())
-        Dim q = Function(e As Integer)
+        Dim q = {|Definition:Function(e As Integer)
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Test2/FindReferences/FindReferencesTests.DelegateInvokeMethodSymbols.vb]: THis test was simply bogus.  The test harness filtered out this span, but we do see this span when doing FAR or highlighting in the IDE.

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide   This will help make it easier to run all our existing tests on the new streaming FAR system that we're introducing.

**Comment by CyrusNajmabadi** [comment]
@dotnet-bot retest windows_eta_open_prtest please
// Previous failure: http://dotnet-ci.cloudapp.net/job/Private/job/dotnet_roslyn-internal/job/master/job/windows_eta_open_prtest/1533/
// Retest reason:

**Comment by Pilchie** [comment]
👍

**Pull request merged**